wallet/src/main.rs

```diff
@@ -160,7 +160,18 @@ fn main() -> Result<(), Box<dyn error::Error>> {
                         .help("The number of lamports to request"),
                 ),
         )
-        .subcommand(SubCommand::with_name("balance").about("Get your balance"))
+        .subcommand(
+            SubCommand::with_name("balance")
+                .about("Get your balance")
+                .arg(
+                    Arg::with_name("pubkey")
+                        .index(1)
+                        .value_name("PUBKEY")
+                        .takes_value(true)
+                        .required(false)
```

Comment:
Try plugging in a clap validator to improve error handling.

Define this somewhere:
```rust
fn pubkey_validator(pubkey: String) -> Result<(), String> {
    match pubkey.parse::<Pubkey>() {
        Ok(_) => Ok(()),
        Err(err) => Err(format!("{:?}", err)),
    }
}
```
then add `.validator(pubkey_validator)` here